pkgs/os-specific/darwin/grandperspective/default.nix

```
@@ -12,9 +12,20 @@ stdenv.mkDerivation (finalAttrs: {
 
   sourceRoot = "GrandPerspective.app";
   buildInputs = [ undmg ];
+  # Create a trampoline script in $out/bin/ because a symlink doesn’t work for
+  # this app. The script must be created at runtime instead of using
+  # e.g. pkgs.writeScript because the variable $out is baked into it, and that’s
+  # only available after the entire derivation has finished evaluating.
   installPhase = ''
-    mkdir -p "$out/Applications/GrandPerspective.app";
-    cp -R . "$out/Applications/GrandPerspective.app";
+    mkdir -p "$out/Applications/GrandPerspective.app" "$out/bin"
+    cp -R . "$out/Applications/GrandPerspective.app"
+    trampoline="$out/bin/grandperspective"
+    >"$trampoline" cat <<EOF
+#!${runtimeShell}
+
+exec '$out/Applications/GrandPerspective.app/Contents/MacOS/GrandPerspective' "\$@"
+EOF
+    chmod 755 "$trampoline"
```

How about, with `makeBinaryWrapper` added to `nativeBuildInputs`:

```suggestion
    makeWrapper "$out/Applications/GrandPerspective.app/Contents/MacOS/GrandPerspective" "$out/bin/grandperspective"
```

---

yeah good idea, didn't think about that 👍